code/modules/mob/living/simple_animal/hostile/terror_spiders/ghost.dm

```diff
@@ -0,0 +1,44 @@
+
+/mob/living/simple_animal/hostile/poison/terror_spider/Topic(href, href_list)
+	if(href_list["activate"])
+		var/mob/dead/observer/ghost = usr
+		if(istype(ghost))
+			humanize_spider(ghost)
+
+/mob/living/simple_animal/hostile/poison/terror_spider/attack_ghost(mob/user)
+	humanize_spider(user)
+
+/mob/living/simple_animal/hostile/poison/terror_spider/proc/humanize_spider(mob/user)
+	if(key)//Someone is in it
+		return
+	var/error_on_humanize = ""
+	var/humanize_prompt = "Take direct control of " + src.name + "? "
+	humanize_prompt += "Role: " + spider_role_summary
+	if(user.ckey in ts_ckey_blacklist)
+		error_on_humanize = "You are not able to control any terror spider this round."
+	else if(!ai_playercontrol_allowingeneral)
+		error_on_humanize = "Terror spiders cannot currently be player-controlled."
+	else if(spider_awaymission)
+		error_on_humanize = "Terror spiders that are part of an away mission cannot be controlled by ghosts."
+	else if(!ai_playercontrol_allowtype)
+		error_on_humanize = "This specific type of terror spider is not player-controllable."
+	else if(health == 0)
+		error_on_humanize = "Dead spiders are not player-controllable."
+	if(jobban_isbanned(user, "Syndicate") || jobban_isbanned(user, "alien"))
+		to_chat(user,"You are jobbanned from role of syndicate and/or alien lifeform.")
+		return
+	if(error_on_humanize == "")
+		var/spider_ask = alert(humanize_prompt, "Join as Terror Spider?", "Yes", "No")
+		if(spider_ask == "No" || !src || qdeleted(src))
+			return
+	else
+		to_chat(user, "Cannot inhabit spider: " + error_on_humanize)
+		return
```

Comment:
You might want to put this check in its own function, so you can poll it elsewhere - perhaps have the check function return a 2-entry list, the first entry being whether it's possible, the second entry being the reason.